proposals/nnnn-remote-control-onRcStatus-notification.md

```diff
@@ -0,0 +1,92 @@
+
+SDL can free the allocated resource for various reasons. In most of the cases, the application knows it no longer has the control of the module. However, in some cases, the application does not know it loses the control of the module. For example, a policy update can revoke the application's access to a module. In another example, when a foreground application tries to access an allocated resource, SDL can de-allocate the module from a background application and re-allocate the module to the foreground application. It happens either automatically (the `RCAccessMode` of RC settings is configured as `AUTO_ALLOW`) or manually with the driver's allowing the control right transfer via a popup (the `RCAccessMode` of RC settings is configured as `ASK_DRIVER`). However, in both examples, the application that previously controls the module does not know the change until it receives a failure response on next control request. The HMI also does not aware the change.
+
+## Proposed solution
+
+Add an `OnRCStatus` notification RPC to `Mobile_API.xml` and `HMI_API.xml`
+
+SDL shall send OnRCStatus notifications to both the mobile application and the HMI whenever SDL allocates a module to an application or it deallocates a module from an application.
+
+#### Existing data related to the change
+```
+<enum name="ModuleType">
+  <element name="CLIMATE"/>
+  <element name="RADIO"/>
+  <element name="AUDIO"/>
+  <element name="LIGHT">
+  <element name="HMI_SETTINGS">
+  <element name="SEAT">
+</enum>
+```
+
+#### Additions to Mobile_API
+
+```xml
+<element name="OnRCStatusID" value="32784" hexvalue="801A" />
+
+<function name="OnRCStatus" messagetype="notification">
+  <description>Issued by SDL to notify the application about remote control status change on SDL</description>
+  <param name="allocatedModules" type="ModuleType" minsize="0" maxsize="100" array="true" mandatory="true">
+    <description>Contains a list (zero or more) of module types that allocated to the application.</description>
+  </param>
+</function>
+```
+Due to there is no agreement on the ID/ZONE scheme of the resources, we use `ModuleType` here. We shall use the id of a resource if we want to support multiple resources of the same type in the future.
+
+#### Additions to HMI_API
+```xml
+<function name="OnRCStatus" messagetype="notification">
+  <description>Issued by SDL to notify HMI about remote control status change on SDL</description>
+  <param name="appID" type="Integer" mandatory="true">
+    <description>ID of selected application.</description>
+  </param>
+  <param name="allocatedModules" type="ModuleType" minsize="0" maxsize="100" array="true" mandatory="true">
+    <description>Contains a list (zero or more) of module types that allocated to the application.</description>
```

Comment:
Please change "...module types that allocated to the application." to "...module types that are allocated to the application."

Reply:
updated